indent-bars.el

```diff
@@ -1238,32 +1238,31 @@ Adapted from `highlight-indentation-mode'."
   (remove-hook 'text-scale-mode-hook #'indent-bars--resize-stipple t)
   (remove-hook 'post-command-hook #'indent-bars--highlight-current-depth t)
   (remove-hook 'font-lock-extend-region-functions
-	       #'indent-bars--extend-blank-line-regions t))
+	       #'indent-bars--extend-blank-line-regions t)
+  (remove-hook 'after-make-frame-functions #'indent-bars-frame-setup t))
 
 (defun indent-bars-reset ()
   "Reset indent-bars config."
   (interactive)
   (indent-bars-teardown)
   (indent-bars-setup))
 
-(defun indent-bars-setup-and-remove ()
-  "Setup indent bars and remove from `after-make-frame-functions'."
-  (remove-hook 'after-make-frame-functions #'indent-bars-setup-and-remove)
-  (indent-bars-setup))
+(defun indent-bars-frame-setup (frame)
+  "Setup indent bars for newly created frame"
+  (select-frame frame)
+  (when (display-graphic-p frame)
+    (indent-bars-setup)))
 
 ;;;###autoload
 (define-minor-mode indent-bars-mode
   "Indicate indentation with configurable bars."
   :global nil
   :group 'indent-bars
   (if indent-bars-mode
-      (if (and (daemonp) (not (frame-parameter nil 'client)))
-	  (let ((buf (current-buffer)))
-	    (add-hook 'after-make-frame-functions
-		      (lambda () (with-current-buffer buf
```

Comment:
This lamba was causing the frame to error and close for me.
First issue was after-make-frame-functions need to take a frame parameter and this takes non, so would fail to get called. 
The other issue is the buf variable didn't seem to get captured as a closure for inside the lambda, so was nil on running with-current-buffer